sdks/java/io/solace/src/main/java/org/apache/beam/sdk/io/solace/broker/BasicAuthJcsmpSessionServiceFactory.java

```diff
@@ -65,11 +64,15 @@ public abstract static class Builder {
 
   @Override
   public SessionService create() {
-    return new BasicAuthJcsmpSessionService(
-        checkStateNotNull(queue, "SolaceIO.Read: Queue is not set.").getName(),
-        host(),
-        username(),
-        password(),
-        vpnName());
+    BasicAuthJcsmpSessionService.Builder builder = BasicAuthJcsmpSessionService.builder();
+    if (queue != null) {
+      builder = builder.queueName(queue.getName());
+    }
```

Comment:
Why did this change from throwing to checking?
Also, given that AutoValue implementations are immutable, deriving from a class that allows you to mutate the queue name seems to run counter to what the framework provides. SessionServiceFactory should probably be an interface declaring create and optionally define an abstract class that implements the interface (abstract of course) and declares equals and hashCode abstract if your intention is to have all implementors explicitly override those methods for value equality.

Reply:
This class was initially used for the `Read` connector alone, which requires that `queue` property to be set, and should fail if it is not set.

The `Write` connector does not need a pre-existing `queue`. I changed the `queueName` parameter to `@Nullable` and this to a check since now the `queueName` value is optional.

Initially, there were different classes for `SessionService`-ish for both connectors, with different properties. Now we use a single class, for consistency across the `Read` and `Write` connectors.